src/panel/direction/DirectionPanel.jsx

```diff
@@ -65,11 +66,27 @@ export default class DirectionPanel extends React.Component {
     this.dragPointHandler = listen('change_direction_point', this.changeDirectionPoint);
     this.setPointHandler = listen('set_direction_point', this.setDirectionPoint);
 
-    if (!this.props.origin && isMobileDevice()) {
-      // If authorized, set origin to current position on mobile
+    // This flag determines if the user's position needs to be requested when opening the direction panel
+    let requestPosition = false;
+
+    // On mobile, and if both origin and destination are empty, check the browser's permissions
+    if (
+      !this.state.origin
+      && !this.state.destination
+      && !this.props.origin
+      && !this.props.destination
+      && isMobileDevice()
+    ) {
+      requestPosition = await Geolocation.showGeolocationModalIfNeeded();
```

Comment:
I thought you were going to change what `showGeolocationModalIfNeeded` does, or at least its name.
Here we don't see clearly how the fact to show the modal (only once, and should be remembered) is related to requesting the position (possibly each time).
Both concepts should be separated.